### DISCUSSION TITLE
hash key state changed before query returns

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#3760] [Q&A]
Hello,

We have a normnal use case where a useQuery key have  in it a state that changes when a user selects something form a dropdown w enoticed that if the user changes options too fast the queries get attached to the wrong state.
In other words the query returns after the state has chnaged and thgere fore we get the wrong results displyaed on screen..
R we doing anything wrong in how we pass the state keys to the hash.
Is there a way to tell use query to only fir if another version of this query is not running .....

**Comment by TkDodo** [comment]
impossible to say without an example of how you are doing things
